workflow/executor/agent.go

```diff
@@ -298,7 +299,15 @@ func (ae *AgentExecutor) executeHTTPTemplateRequest(ctx context.Context, httpTem
 		}
 		request.Header.Add(header.Name, value)
 	}
+
 	httpClient := http.DefaultClient
+	if httpTemplate.InsecureSkipVerify != nil {
+		transCfg := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: *httpTemplate.InsecureSkipVerify},
+		}
+		httpClient = &http.Client{Transport: transCfg}
```

Comment:
hmmmm.....

Performance issue here, we re-use the DefaultClient, so things like keep-alives will work well. 

Instead, great two global clients:

```golang
var httpClients = map[bool]http.Client {
false: http.DefaultClient.Clone().SetInsecureSkipVerify(false),
true: http.DefaultClient.Clone().SetInsecureSkipVerify(true)
}
```

The get the client you need from that map.

Reply:
For false, could we return `*http.DefaultClient` and for true, `http.Client{Transport: transCfg} as `http.DefaultClient' doesn't have Clone()

Reply:
Sure, I think that would be OK too. Key point - reuse the client.

Reply:
Got it, think current implementation is correct then.